src/tools/compiletest/src/runtest.rs

```diff
@@ -3258,6 +3266,45 @@ impl<'test> TestCx<'test> {
         let stamp = crate::stamp(&self.config, self.testpaths, self.revision);
         fs::write(&stamp, compute_stamp_hash(&self.config)).unwrap();
     }
+
+    fn run_yk_tir_test(&self) {
+        let proc_res = self.compile_test();
+
+        if !proc_res.status.success() {
+            self.fatal_proc_rec("compilation failed!", &proc_res);
+        }
+
+        if !proc_res.status.success() {
+            self.fatal_proc_rec("test run failed!", &proc_res);
+        }
+
+        self.check_yk_tir_dump();
+    }
+
+    fn check_yk_tir_dump(&self) {
+        let test_file_contents = fs::read_to_string(&self.testpaths.file).unwrap();
+        if let Some(idx) = test_file_contents.find("// END RUST SOURCE") {
+            let (_, test_text) = test_file_contents.split_at(idx + "// END_RUST SOURCE".len());
+            let mut test_lines = vec![ExpectedLine::Elision];
+            for l in test_text.lines() {
+                if l.is_empty() {
+                    // ignore
+                } else if l.starts_with("//") && l.split_at("//".len()).1.trim() == "..." {
```

Comment:
This feels like it would be clearer (and faster) as something like:

```rust
if l.starts_with("//") {
   if l["".len()..].trim() == "..." { ... }
   else { ... }
}
```

Reply:
As before, let's follow upstream's example.